inference-engine/src/preprocessing/ie_preprocess_gapi.cpp

```diff
@@ -977,16 +1153,14 @@ void PreprocEngine::preprocessBlob(const BlobTypePtr &inBlob, MemoryBlob::Ptr &o
         omp_serial, update);
 }
 
-void PreprocEngine::preprocessWithGAPI(const Blob::Ptr &inBlob, Blob::Ptr &outBlob,
-        const ResizeAlgorithm& algorithm, ColorFormat in_fmt, bool omp_serial, int batch_size) {
-    const auto out_fmt = (in_fmt == ColorFormat::RAW) ? ColorFormat::RAW : ColorFormat::BGR;  // FIXME: get expected color format from network
-
-    // output is always a memory blob
-    auto outMemoryBlob = as<MemoryBlob>(outBlob);
-    if (!outMemoryBlob) {
-        IE_THROW()  << "Unsupported network's input blob type: expected MemoryBlob";
-    }
+void PreprocEngine::preprocessBatchedBlob(const BatchedBlob::Ptr &inBlob, MemoryBlob::Ptr &outMemoryBlob,
+        ResizeAlgorithm algorithm, ColorFormat in_fmt, ColorFormat out_fmt, bool omp_serial,
+        int batch_size) {
+    preprocessTypedBlob(inBlob, outMemoryBlob, algorithm, in_fmt, out_fmt, omp_serial, batch_size);
```

Comment:
So what is this and how does it work?

Reply:
let me remember...
most of the code had been written to satisfy legacy behavior and reuse it for BacthedBlob
in high level `preprocessWithGAPI` determine Blob type  and call 
- or `preprocessBatchedBlob` with come `batch_size` correcting which underlying `preprocessTypedBlob` and preprocessBlob
- or `preprocessTypedBlob` directly without correction

I'll explain later when I recall it by myself

Reply:
yeah but this function calls `preprocessTypedBlob` as well - so why is it required here?

Reply:
My point is that - I lost in this review and couldn't find the actual place where batched blob is preprocessed - can you please post a link here? Thanks

Reply:
I checked existing code and call-stack looks like (up to down)

```
-preprocessWithGAPI
/* legacy */
----preprocessBlob (uses input fmt switch-case and type extraction using dynamic_cast to provide sanity check)
--------preprocessTypedBlob <NV12, I420, MemoryBlob>
-----------<some unified-generic code>
------------------bind_to_blob<NV12, I420, Memory>
/ *BB */
----preprocessBatchedBlob (does not use any fmt and type extraction)
--------preprocessTypedBlob <BatchedBlob>
-----------<some unified-generic code>
------------------bind_to_blob<BatchedBlob>(type extraction happens here)
----------------------bind_to_blob<NV12, I420, Memory>
                                         
```

I see that after several refactoring (and generalizations) it is possible either to unify `preprocessBlob ` and `preprocessBatchedBlob ` into single one or rename  `preprocessBlob ` to `preprocessSingleBlob` as well with current behavior. @dmatveev  what do you suggest ?